hipamd/include/hip/amd_detail/amd_warp_sync_functions.h

```diff
@@ -224,17 +224,37 @@ unsigned long long __match_any(T value) {
           (sizeof(T) == 4 || sizeof(T) == 8),
       "T can be int, unsigned int, long, unsigned long, long long, unsigned "
       "long long, float or double.");
-  bool done = false;
-  unsigned long long retval = 0;
 
-  while (__any(!done)) {
-    if (!done) {
-      T chosen = __hip_readfirstlane(value);
-      if (chosen == value) {
-        retval = __activemask();
-        done = true;
+  unsigned long long actvmask = __activemask();
+  unsigned long long retval = 0;
+  if (actvmask != ~0ull) {
+    bool done = false;
+    while (__any(!done)) {
+      if (!done) {
+        T chosen = __hip_readfirstlane(value);
+        if (chosen == value) {
+          retval = __activemask();
+          done = true;
+        }
       }
     }
+  } else {
+    union dill { unsigned int i[2]; unsigned long long ill; decltype(value) val; } dill_ = { .val = value };
+    retval = 1;
+    //Do a full rotate of the wave lanes, using dpp with "wave_rol1" control (ID: 0x134).
```

Comment:
```suggestion
    // Do a full rotate of the wave lanes, using dpp with "wave_rol1" control (ID: 0x134).
```
Please add a space after the `//` for every comment line. I don't know if HIP has yet adopted a coding style, but this kind of whitespace is very commonly considered correct.